ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```diff
@@ -0,0 +1,287 @@
+from hashlib import sha256
+
+from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
+from ansible_collections.arista.avd.plugins.filter.esi_management import generate_esi, generate_route_target
+from ansible_collections.arista.avd.plugins.plugin_utils.merge import merge
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get, get_item
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.interface_descriptions import AvdInterfaceDescriptions
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.ip_addressing import AvdIpAddressing
+
+
+class UtilsMixin:
+    """
+    Mixin Class with internal functions.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    # Set type hints for Attributes of the main class as needed
+    _hostvars: dict
+    _avd_ip_addressing: AvdIpAddressing
+    _avd_interface_descriptions: AvdInterfaceDescriptions
+
+    @cached_property
+    def _filtered_connected_endpoints(self) -> list:
+        """
+        Return list of endpoints defined under one of the keys in "connected_endpoint_keys"
+        which are connected to this switch.
+
+        Adapters are filtered to contain only the ones connected to this switch.
+        """
+        filtered_connected_endpoints = []
+        for connected_endpoints_key in self._connected_endpoints_keys:
+            connected_endpoints = convert_dicts(get(self._hostvars, connected_endpoints_key["key"], default=[]), "name")
+            for connected_endpoint in connected_endpoints:
+                if "adapters" not in connected_endpoint:
+                    continue
+
+                connected_endpoint["adapters"] = [adapter for adapter in connected_endpoint["adapters"] if self._hostname in adapter.get("switches", [])]
+
+                for adapter_index, adapter in enumerate(connected_endpoint["adapters"]):
+                    if "profile" not in adapter:
+                        continue
+
+                    port_profile = get_item(self._merged_port_profiles, "profile", adapter["profile"], default={})
+                    adapter = merge(port_profile, adapter, list_merge="replace", destructive_merge=False)
+                    adapter.pop("profile")
+                    connected_endpoint["adapters"][adapter_index] = adapter
+
+                connected_endpoint["type"] = connected_endpoints_key["type"]
+                filtered_connected_endpoints.append(connected_endpoint)
+
+        return filtered_connected_endpoints
+
+    @cached_property
+    def _filtered_network_ports(self) -> list:
+        """
+        Return list of endpoints defined under "network_ports"
+        which are connected to this switch.
+        """
+        filtered_network_ports = []
+        for network_port in get(self._hostvars, "network_ports", default=[]):
+            if not self._match_regexes(network_port.get("switches"), self._hostname):
```

Comment:
can probably use Python `any` here https://docs.python.org/3/library/functions.html#

```suggestion
if any(re.match(rf"^{regex}$", self._hostname) for regex in network_port.get("switches")):
```

Reply:
I have updated the helper function. We can consider moving it to utils at some point when we broaden support for matching like this.